src/CompareRevisions/GitHub.hs

```diff
@@ -25,22 +26,34 @@ type Organization = Text
 -- | The name of a repository.
 type RepositoryName = Text
 
--- | Assuming a Git URL points to a GitHub repository, generate a URL for that
--- repository's web page.
-websiteURI :: Git.URL -> Maybe URI
-websiteURI url@(Git.URI uri) =
-  Just $ uri { uriPath = toS (repoPath url) }
-websiteURI url@(Git.SCP scp) =
-  foreach (SCP.getHostname scp) $ \hostname ->
-    URI.nullURI
-    { uriScheme = "https:"
-    , uriAuthority = Just URI.URIAuth
-      { URI.uriUserInfo = ""
-      , URI.uriRegName = toS (SCP.unHostname hostname)
-      , URI.uriPort = ""
-      }
-    , uriPath = toS (repoPath url)
+-- | A repository on GitHub.
+--
+-- Construct directly, or from a Git URL with 'getRepo'.
+data Repository
+  = Repository
+  { organization :: Organization -- ^ The name of the organization.
+  , repositoryName :: RepositoryName -- ^ The name of the repository.
+  } deriving (Eq, Show)
+
+-- | Get the GitHub organization and repository from a Git URL.
+repositoryFromGitURL :: Git.URL -> Maybe Repository
+repositoryFromGitURL uri =
```

Comment:
Could make this check that the hostname is github.com

Reply:
Definitely could! Was actually thinking of storing the hostname in the struct. Going to merge as is for now.

Reply:
:+1: